internal/provider/kubernetes/predicates_test.go

```diff
@@ -356,6 +356,9 @@ func TestValidateSecretForReconcile(t *testing.T) {
 	r := gatewayAPIReconciler{
 		classController: egv1a1.GatewayControllerName,
 		log:             logger,
+		spCRDExists:     true,
```

Comment:
I'm think about somehting like `sets[GVK]` instead of introduce two much properties.

Reply:
yeah, the sets approach saves a few lines (~10) of code by not instantiating the `bool`, but will add the same amount when inserting values into the set

Reply:
I dont see this changes much in the future, imo, the APIs have kinda stabilized